easybuild/framework/easyblock.py

```diff
@@ -514,10 +547,19 @@ def fetch_extension_sources(self, skip_checksums=False):
 
                     if ext_options.get('nosource', None):
                         exts_sources.append(ext_src)
+                    elif ext_options.get('sources', None):
+                        sources = ext_options['sources']
+                        src = self.fetch_source(sources, checksums, extension=True)
+                        # Copy 'path' entry to 'src' for use with extensions
+                        ext_src.update({'src': src['path']})
+                        exts_sources.append(ext_src)
                     else:
                         source_urls = ext_options.get('source_urls', [])
                         force_download = build_option('force_download') in [FORCE_DOWNLOAD_ALL, FORCE_DOWNLOAD_SOURCES]
-                        src_fn = self.obtain_file(fn, extension=True, urls=source_urls, force_download=force_download)
+                        git_config = ext_options.get('git_config', None)
```

Comment:
This doesn't make much sense to me, to support `git_config` outside of `sources`.

In the long run, we want to deprecate the use of `source_tmpl`, and replace it with just using `sources`, which is more consistent with the top-level `sources` easyconfig parameter. There is no `git_config` easyconfig parameter either, it only exists in the context of `sources`.

Reply:
Yes, this makes sense.  The idea that each `exts_list` entry is like a miniature easyconfig, immediately implies that it shouldn't have a top-level `git_config`.  I plan to add this explanation to the documentation (docs PR #[617](https://github.com/easybuilders/easybuild/pull/617)).